### PR TITLE
Feature/chat channel

### PR DESCRIPTION
**Pull request by smikulas**

### CODE REVIEW

File: app/src/main/java/com/github/gogetters/letsgo/chat/model/ChatMessageData.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    val toId: String,
+    val sendTime: Long
+) {
+
+    /**
+     * Empty constructor is required to parse the object
+     */
+    constructor() : this("", "", "", "", -1)
+
+    fun toMap(): Map<String, Any?> {
```
Comment (sirpudika) [inline app/src/main/java/com/github/gogetters/letsgo/chat/model/ChatMessageData.kt:19]: Easy to test

File: app/src/main/java/com/github/gogetters/letsgo/chat/model/UserData.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
+/**
+ * The UserData class represents the users registered and fetched from the database
+ * It implements the Parcelable interface to be passed as object between activities
+ */
+@Parcelize
+data class UserData(
```
Comment (sirpudika) [inline app/src/main/java/com/github/gogetters/letsgo/chat/model/UserData.kt:11]: Easy to test, pls do it

**Review by sirpudika — COMMENTED** [review]
Your written code looks really good, thanks a lot for that! Unfortunately, we cannot merge it like that as it takes the test coverage down by 4.4% and there is a lot of code that can easily been tested. We can help you with that, let's discuss tonight.

File: app/src/main/java/com/github/gogetters/letsgo/activities/ChatActivity.kt [hunk 3]
```diff
@@ -1,151 +1,112 @@
...
-
-    // connect to the Chat Collection's Message Document in Firestore
-//    private val firestoreChat by lazy {
-//        FirebaseFirestore.getInstance().collection(COLLECTION_KEY).document(DOCUMENT_KEY)
-//    }
-
-    // data structure in Firestore
+    val adapter = GroupAdapter<ViewHolder>()
+    var userId: String? = null
+    var toUser: UserData? = null
```
Comment (kprotopapas) [inline app/src/main/java/com/github/gogetters/letsgo/activities/ChatActivity.kt:24]: what is this variable for? it is not clear from the name imo

File: app/src/main/java/com/github/gogetters/letsgo/activities/ChatActivity.kt [hunk 4]
```diff
@@ -1,151 +1,112 @@
...
+
     companion object {
-        const val COLLECTION_KEY = "Chat"
-        const val DOCUMENT_KEY = "Message"
-        const val NAME_FIELD = "Name"
-        const val TEXT_FIELD = "Text"
+        val UNKNOWN = "Unknown"
```
Comment (kprotopapas) [inline app/src/main/java/com/github/gogetters/letsgo/activities/ChatActivity.kt:27]: this as well, if it is used for the userId maybe it should have some reference to the ID in the name

File: app/src/main/java/com/github/gogetters/letsgo/activities/ChatLastMessageActivity.kt [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+import com.google.firebase.database.FirebaseDatabase
+import com.xwray.groupie.GroupAdapter
+import com.xwray.groupie.ViewHolder
+import kotlinx.android.synthetic.main.activity_chat_last_message.*
+
+/**
+ * Lists last messages previously received from users
+ * Redirects to ChatNewMessageActivity
+ */
+class ChatLastMessageActivity : AppCompatActivity() {
```
Comment (kprotopapas) [inline app/src/main/java/com/github/gogetters/letsgo/activities/ChatLastMessageActivity.kt:24]: why are last messages important?

**Review by kprotopapas — APPROVED** [review]
same comments as Jan, some stuff could be tested but otherwise great test coverage. Also some variable names I didnt' understand but maybe I am just stupid

**Review by lupusmalus — APPROVED** [review]
Nothing to add to the prior comments

**Review by sirpudika — APPROVED** [review]

**Pull request merged**